pkg/wekafs/apistore.go

```diff
@@ -54,43 +60,44 @@ func (api *ApiStore) getByClusterGuid(guid uuid.UUID) (*apiclient.ApiClient, err
 
 // fromSecrets returns a pointer to API by secret contents
 func (api *ApiStore) fromSecrets(ctx context.Context, secrets map[string]string, hostname string) (*apiclient.ApiClient, error) {
-	endpointsRaw := strings.TrimSpace(strings.ReplaceAll(strings.TrimSuffix(secrets["endpoints"], "\n"), "\n", ","))
+	endpointsRaw, ok := secrets["endpoints"]
+	if !ok {
+		return nil, fmt.Errorf("no endpointsRaw found in secret")
+	}
+	endpointsRaw = strings.ReplaceAll(cleanSecretValue(endpointsRaw), "\n", ",")
 	if endpointsRaw == "" {
 		return nil, errors.New("no valid endpoints defined in secret, cannot create API client")
```

Comment:
I think you should return status.Errorf() instead with a valid mapping to Kubernetes error code.
Unless this is wrapped later on